crates/pipeline_manager/src/retries.rs

```diff
@@ -0,0 +1,25 @@
+use std::{future::Future, time::Duration};
+
+pub async fn retry_async<T, E, O, F: FnMut() -> O>(
+    mut f: F,
+    retries: i32,
+    interval: Duration,
```

Comment:
maybe client should also be able to provide a message that gets printed as a warning after a few failed tries

Reply:
I thought about it but it can be done with more context from within the passed FnMut itself (by looking at the error type etc).